src/main/java/seedu/address/ui/CommandBox.java

```diff
@@ -43,12 +51,72 @@ private void handleCommandEntered() {
 
         try {
             commandExecutor.execute(commandText);
+            commandHistory.add(commandText);
+            commandHistoryPointerIndex = commandHistory.size();
             commandTextField.setText("");
         } catch (CommandException | ParseException e) {
             setStyleToIndicateCommandFailure();
         }
     }
 
+    /**
+     * Handles arrow key (up and down) pressed event.
+     */
+    @FXML
+    private void handleArrowKeyPressed(KeyEvent keyEvent) {
+        KeyCode keyPressed = keyEvent.getCode();
+        switch (keyPressed) {
+        case UP:
+            keyEvent.consume();
+            showPreviousCommand();
+            break;
+        case DOWN:
+            keyEvent.consume();
+            showNextCommand();
+            break;
+        default:
+        }
+    }
+
+    private void showPreviousCommand() {
+        if (commandHistoryPointerIndex < 0) {
+            return;
+        }
+        try {
+            commandHistoryPointerIndex--;
+            String previousCommand = commandHistory.get(commandHistoryPointerIndex);
+            setCommandBoxText(previousCommand);
+        } catch (IndexOutOfBoundsException e) {
+            return; //Nothing needs to be done for requests beyond what is stored in history.
```

Comment:
Just for consistency, although it's not an important change.
```suggestion
            return; // Nothing needs to be done for requests beyond what is stored in history.
```